parsl/executors/high_throughput/zmq_pipes.py

```diff
@@ -37,8 +55,21 @@ def run(self, message):
         in ZMQ sockets reaching a broken state once there are ~10k tasks in flight.
         This issue can be magnified if each the serialized buffer itself is larger.
         """
-        self.zmq_socket.send_pyobj(message, copy=True)
-        reply = self.zmq_socket.recv_pyobj()
+        try:
+            self.zmq_socket.send_pyobj(message, copy=True)
+            reply = self.zmq_socket.recv_pyobj()
+        except zmq.ZMQError:
+            logger.error("Potential ZMQ REQ-REP deadlock caught")
+            logger.info("Trying to reestablish context")
+            self.context.destroy()
+            self.context = zmq.Context()
+            self.port = self.create_socket_and_bind()
+            if max_retries > 0:
```

Comment:
should this retries test be here? or at the start of run() or before line 66 which attempts a reconnect?